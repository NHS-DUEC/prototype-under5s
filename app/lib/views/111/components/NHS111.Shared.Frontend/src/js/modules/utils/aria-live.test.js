const { ariaPoliteMessage, ariaAssertiveMessage } = require("./aria-live");

const ariaRemovalDelay = 110;

describe("aria helpers", () => {
  beforeEach(() => {
    document.body.innerHTML = `
      <div id="aria-assertive"></div>
      <div id="aria-polite"></div>
    `;
  });
  describe("ariaPoliteMessage", () => {
    test("correctly inserts the message into the dom", (done) => {
      ariaPoliteMessage("Test one");

      expect(document.body.innerHTML).toMatchSnapshot("message present");

      setTimeout(() => {
        expect(document.body.innerHTML).toMatchSnapshot("message removed");
        done();
      }, ariaRemovalDelay);
    });
  });
  describe("ariaAssertiveMessage", () => {
    test("correctly inserts the message into the dom", (done) => {
      ariaAssertiveMessage("Test one");

      expect(document.body.innerHTML).toMatchSnapshot("message present");

      setTimeout(() => {
        expect(document.body.innerHTML).toMatchSnapshot("message removed");
        done();
      }, ariaRemovalDelay);
    });
  });
});
