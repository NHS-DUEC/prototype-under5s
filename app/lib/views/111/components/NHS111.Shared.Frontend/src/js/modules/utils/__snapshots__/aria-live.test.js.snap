// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`aria helpers ariaAssertiveMessage correctly inserts the message into the dom: message present 1`] = `
"
      <div id="aria-assertive"><p>Test one</p></div>
      <div id="aria-polite"></div>
    "
`;

exports[`aria helpers ariaAssertiveMessage correctly inserts the message into the dom: message removed 1`] = `
"
      <div id="aria-assertive"></div>
      <div id="aria-polite"></div>
    "
`;

exports[`aria helpers ariaPoliteMessage correctly inserts the message into the dom: message present 1`] = `
"
      <div id="aria-assertive"></div>
      <div id="aria-polite"><p>Test one</p></div>
    "
`;

exports[`aria helpers ariaPoliteMessage correctly inserts the message into the dom: message removed 1`] = `
"
      <div id="aria-assertive"></div>
      <div id="aria-polite"></div>
    "
`;
